### PR TITLE
Variables: adds onTimeRangeUpdated to newVariables

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Relates to #22564

**Special notes for your reviewer**:
I think I've moved this correctly to the new variables but I'm not sure about all the particulars, tried to read up on this as well https://grafana.com/docs/grafana/latest/reference/templating/#interval-variables. Would be awesome if someone could test this together with the Interval variable type.

### CODE REVIEW

File: public/app/features/variables/state/actions.test.ts [hunk 1]
```diff
@@ -267,4 +281,161 @@ describe('shared actions', () => {
...
+        await onTimeRangeUpdated(range, dependencies)(dispatchMock, getStateMock, undefined);
+
+        expect(dispatchMock).toHaveBeenCalledTimes(0);
+        expect(getStateMock).toHaveBeenCalledTimes(4);
+        expect(updateTimeRangeMock).toHaveBeenCalledTimes(1);
+        expect(updateTimeRangeMock).toHaveBeenCalledWith(range);
+        expect(templateVariableValueUpdatedMock).toHaveBeenCalledTimes(1);
+        expect(startRefreshMock).toHaveBeenCalledTimes(1);
+        expect(emitMock).toHaveBeenCalledTimes(0);
+      });
```
Comment (hugohaggmark) [inline public/app/features/variables/state/actions.test.ts]: Not sure about the value of these tests, maybe these tests should be e2e tests instead.

File: public/app/features/variables/state/helpers.ts [hunk 2]
```diff
@@ -164,3 +165,13 @@ export const getTemplatingAndLocationRootReducer = () =>
...
+
+export const getTemplatingAndDashboardReducer = () =>
+  combineReducers({
+    templating: combineReducers({
+      optionsPicker: optionsPickerReducer,
+      editor: variableEditorReducer,
+      variables: variablesReducer,
+    }),
+    dashboard: dashboardReducer,
+  });
```
Comment (hugohaggmark) [inline public/app/features/variables/state/helpers.ts]: Unused, I'll remove this soon.

File: public/app/features/variables/state/actions.test.ts (hunk 1 above)
Comment (torkelo) [reply public/app/features/variables/state/actions.test.ts]: yea, if we have a testdata that can have variables that update based on time we could have an e2e test for it: 

Created this PR that adds a dashboard that could be used in such an e2e test:
https://github.com/grafana/grafana/pull/22829

**Review by torkelo — APPROVED** [review]
Looks good, created a dashboard to test variables that change on time range change: 
https://github.com/grafana/grafana/pull/22829

File: public/app/features/variables/state/actions.ts [hunk 3]
```diff
@@ -323,6 +327,44 @@ export const variableUpdated = (identifier: VariableIdentifier, emitChangeEvents
...
+      dashboard?.templateVariableValueUpdated();
+    }
+  });
+
+  try {
+    await Promise.all(promises);
+    const dashboard = getState().dashboard.getModel();
+    dashboard?.startRefresh();
+  } catch (error) {
+    dependencies.appEvents.emit(AppEvents.alertError, ['Template variable service failed', error.message]);
```
[the remark this answers is not in the document]
Comment (hugohaggmark) [reply public/app/features/variables/state/actions.ts]: Great feedback, I will fix that.

**Pull request merged**